harness/determined/cli/trial.py

```diff
@@ -121,6 +126,77 @@ def trial_logs(args: Namespace) -> None:
     )
 
 
+@authentication.required
+def generate_support_bundle(args: Namespace):
+    temp_dir = tempfile.TemporaryDirectory()
+    trial_logs_filepath = write_trial_logs(args, temp_dir)
+
+    master_logs_filepath = write_master_logs(args, temp_dir)
+
+    api_experiment_filepath, api_trail_filepath = write_api_call(args, temp_dir)
+
+    bundle = None
+    if args.output_dir is None:
+        output_dir = os.getcwd()
+
+    else:
+        bundle = tarfile.open(f"{args.output_dir}.tar.gz", "w:gz")
+
+    filename_printf = "det-bundle-trial-{trial_id}-{datetime}.tar.gz"
+    tar_filename = filename_printf.format(
+        trial_id=args.trial_id, datetime=datetime.now().strftime("%Y%m%dT%H%M%S")
+    )
+    fullpath = os.path.join(output_dir, tar_filename)
+    bundle = tarfile.open(fullpath, "w:gz")
+
+    bundle.add(trial_logs_filepath, arcname=os.path.basename(trial_logs_filepath))
+    bundle.add(master_logs_filepath, arcname=os.path.basename(master_logs_filepath))
+    bundle.add(api_trail_filepath, arcname=os.path.basename(api_trail_filepath))
+    bundle.add(api_experiment_filepath, arcname=os.path.basename(api_experiment_filepath))
+
+    print(f"bundle directory: {bundle.name}")
+
+    bundle.close()
+
+
+def write_trial_logs(args, temp_dir):
+    trial_logs = api.trial_logs(args.master, args.trial_id)
+    file_path = os.path.join(temp_dir.name, "trial_logs.txt")
+    with open(file_path, "w") as f:
+        for log in trial_logs:
+            f.write(log["message"])
+
+    return file_path
+
+
+def write_master_logs(args, temp_dir):
+    response = api.get(args.master, "logs")
+    file_path = os.path.join(temp_dir.name, "master_logs.txt")
+    with open(file_path, "w") as f:
+        for log in response.json():
+            f.write("{} [{}]: {}".format(log["time"], log["level"], log["message"]))
+    return file_path
+
+
+def write_api_call(args, temp_dir):
+    file_path1 = os.path.join(temp_dir.name, "api_experiment_call.json")
+    file_path2 = os.path.join(temp_dir.name, "api_trial_call.json")
+
+    trial_obj = bindings.get_GetTrial(setup_session(args), trialId=args.trial_id).trial
+    experiment_id = trial_obj.experimentId
+    exp_obj = bindings.get_GetExperiment(setup_session(args), experimentId=experiment_id)
+
+    trial_obj.to_json(), exp_obj.to_json()
```

Comment:
I don't think this does anything?

Reply:
yeah that wasn't supposed to be there, thanks!